### PR TITLE
Crowdstrike apimodule

### PR DESCRIPTION
**Pull request by guyfreund**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/26185

## Description
ApiModule to CrowdStrike OAuth2 authentication

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by guyfreund** [comment]
adding the docs approved label since the yml and readme files are copied entirely from MicrosoftApiModule.

File: Packs/ApiModules/Scripts/CrowdStrikeApiModule/CrowdStrikeApiModule.py [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        headers = {
+            'Authorization': f'Basic {base64.b64encode(byte_creds).decode()}'
+        }
+        token_res = self.http_request('POST', '/oauth2/token', data=body, headers=headers)
+        return token_res.get('access_token')
+
+    def check_quota_status(
+            self
+    ) -> dict:
+        """Creating the needed arguments for the http request
```
Comment (ShahafBenYakir) [inline Packs/ApiModules/Scripts/CrowdStrikeApiModule/CrowdStrikeApiModule.py:182]: check_quota_status = Creating the needed arguments for the http request?

File: Packs/ApiModules/Scripts/CrowdStrikeApiModule/CrowdStrikeApiModule.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        self._proxy = params.get('proxy', False)
+        self._base_url = "https://api.crowdstrike.com/"
+        self._ok_codes = tuple()  # type: ignore[var-annotated]
+        self._session = requests.Session()
+        self._token = self._generate_token()
+        self._headers = {'Authorization': 'bearer ' + self._token}
+        if not self._proxy:
+            self._session.trust_env = False
+
+        # super().__init__(base_url=self._base_url, verify=self._verify, *args, **kwargs)  # type: ignore[misc]
```
Comment (ShahafBenYakir) [inline Packs/ApiModules/Scripts/CrowdStrikeApiModule/CrowdStrikeApiModule.py:26]: You wrote in your client declaration that you are inheriting from the BaseClient but you never did call it. Why are not using it and instead we are coping stuff out of it?

File: Packs/Base/Scripts/CommonServerPython/CommonServerPython.py [hunk 3]
```diff
@@ -3709,7 +3714,10 @@ def _http_request(self, method, url_suffix, full_url=None, headers=None, auth=No
                     try:
                         # Try to parse json error response
                         error_entry = res.json()
-                        err_msg += '\n{}'.format(json.dumps(error_entry))
+                        if error_handler:
+                            err_msg += error_handler(error_entry)
+                        else:
+                            err_msg += '\n{}'.format(json.dumps(error_entry))
```
Comment (ShahafBenYakir) [inline Packs/Base/Scripts/CommonServerPython/CommonServerPython.py:3720]: If you want to have the option to use some sort of error handler then put it right under `if not self._is_status_code_valid(res, ok_codes):`
Such that 
```
if not self._is_status_code_valid(res, ok_codes):
    if error_handler:
        return error_handler(res)
```
```

**Comment by guyfreund** [comment]
@kirbles19 please review the yml, readme & changelog files

File: Packs/Base/ReleaseNotes/1_1_8.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### CommonServerPython
+- added *error_handler* argument to BaseClient's http_request method
```
Comment (ShahafBenYakir) [inline Packs/Base/ReleaseNotes/1_1_8.md:4]: Can you also update this in docs? and maybe also in content-team?

**Comment by ShahafBenYakir** [comment]
Bear in mind that Ut is failing

**Pull request merged**